src/firefly/js/visualize/projection/Wavelength.js

```diff
@@ -24,22 +26,102 @@ export const VRAD = 'VRAD';
- * @prop {Array.<number>} pc_3j
- * @prop {number} restWAV
- * @prop {number} crpix
- * @prop {number} cdelt
- * @prop {number} crval
+ * @typedef {Object} SpectralWCSData
+ *
+ * @prop {Array.<SpectralCoord>} spectralCoords  parameters specific to each spectral coordinate
+ * @prop {string} failReason
+ * @prop {boolean} hasPlainOnlyCoordInfo
+ * @prop {boolean} hasPixelLevelCoordInfo
+ * @prop {boolean} isNonSeparableTABGroup all spectral coordinates are TAB that should be processed together
  */
 
 
 
-const wlTypes = {
+
+/**
+ * @global
+ * @public
+ * @typedef {Object} SpectralCoord
+ * @summary The information needed to calculate and present (public interface) spectral coordinate value
+ *
+ * @prop {boolean} planeOnly     coordinate can be calculated per plane (public interface)
+ * @prop {string} name           spectral coordinate name (public interface)
+ * @prop {string} symbol         spectral coordinate symbol (public interface)
+ * @prop {string} units          spectral coordinate units (public interface)
+ * @prop {string} coordType      spectral coordinate type
+ * @prop {string} algorithm      the algorithm used to calculate spectral coordinate value
+ * @prop {string} ctype          spectral coordinate type in the form XXXX-YYY, where XXXX is coordinate type, YYY is algorithm
+ * @prop {Number} N              (WCS level) the dimensionality of WCS representation given either by NAXIS or WCSAXES
+ * @prop {Array.<number>} r_j    (WCS Level) pixel coordinates of the reference point given by CRPIXj
+ * @prop {Array.<number>} pc_3j  the linear transformation matrix row corresponding to the spectral coordinate i (given either by PCi_j or CDi_j)
+ * @prop {number} cdelt          scaling factor CDELTi or 1.0
+ * @prop {number} crval          spectral coordinate value at reference point
+ * @prop {number} [restWAV]      (WCS Level) rest wavelength (used for V2W algorithm)
+ * @prop {string} [ps3_1]        column name for the coordinate array (PSi_1a in table lookup algorithm)
+ * @prop {string} [ps3_2]        column name for indexing vector (PSi_2a in table lookup algorithm)
+ * @prop {int} [pv3_3]           axis number (m) in array PSi_1a (table lookup algorithm)
+ * @prop {Table} [table]         lookup table (table lookup algorithm)
+ */
+
+
+/**
+ * Spectral-coordinate type codes.
+ */
+export const spectralCoordTypes = {
+    ['FREQ']: {
+        type: 'Frequency',
+        symbol: String.fromCharCode(0x03BD), // ν (nu)
+        defaultUnits: 'Hz',
+    },
+    ['ENER']: {
+        type: 'Energy',
+        symbol: 'E',
+        defaultUnits: 'J',
+    },
+    ['WAVN']: {
+        type: 'Wavenumber',
+        symbol: 'k',
+        defaultUnits: '1/m',
+    },
+    ['VRAD']: {
+        type: 'Radio velocity',
+        symbol: 'V',
+        defaultUnits: 'm/s',
+    },
+    ['WAVE']: {
+        type: 'Vacuum wavelength',
+        symbol: String.fromCharCode(0x03BB), // λ
+        defaultUnits: 'm',
+    },
```

Comment:
Should we add a list of aliases for each item here? It would contain a list of coordinate types that should be treated as the known spectral type even though their CTYPE is not quite conforming to the standard.